src/app/view/components/_common/site-selector/site-selector.component.ts

```diff
@@ -85,6 +85,19 @@ export class SiteSelectorComponent implements OnInit, OnChanges {
      */
     handleSitesRefresh(): void {
         this.paginationService.getCurrentPage().subscribe((items) => {
+            const sites: any[] = items.splice(0);
+            this.getSitesList();
+
+            // if current site not being listed on updated sites list (socket message) then switch to default site
+            if (!sites.find((site) => site.identifier === this.siteService.currentSite.identifier)) {
```

Comment:
This conditions is too complicated, when we find this kind of conditions we move it into a private method with a name that tells what is doing.

Reply:
done